torch/distributed/_composable/fsdp/_fsdp_api.py

```diff
@@ -0,0 +1,45 @@
+from dataclasses import dataclass
+from typing import Optional
+
+import torch
+
+
+@dataclass(frozen=True)
+class MixedPrecisionPolicy:
+    """
+    This configures FSDP's mixed precision. Unlike autocast, this applies mixed
+    precision at the module level, not op level, which means low-precision
+    activations are saved for backward and high-to-low-precision casts are
+    incurred only at module boundaries.
+
+    FSDP works well with module-level mixed precision since it keeps the
+    high-precision sharded parameters in memory anyway. In other words, FSDP
+    does not require any extra memory to keep a high-precision copy of the
+    parameters for the optimizer step.
+
+    Attributes:
+        param_dtype (Optional[torch.dtype]): This specifies the dtype for
```

Comment:
I am actually interesting about how the mixed dtype would work here, what if user want to specify some modules with a dtype but all other modules with another dtype? the param_dtype would ignore the "some modules"?

Reply:
Yes, almost like that! The current plan is for mixed dtype to be a tensor-level extension that overrides the module-level mixed precision policy.